src/reconstruction/enzyme_constrained.jl

```diff
@@ -42,6 +42,15 @@ function make_enzyme_constrained_model(
     gene_product_mass_group_bounds::Maybe{Dict{String,Float64}} = nothing,
     total_gene_product_mass_bound::Maybe{Float64} = nothing,
 )
+    all(
+        !isnothing,
+        [
+            gene_product_mass_groups,
+            gene_product_mass_group_bounds,
+            total_gene_product_mass_bound,
+        ],
+    ) && throw(ArgumentError("Too many arguments specified!"))
+
     if !isnothing(total_gene_product_mass_bound)
         gene_product_mass_groups = Dict("uncategorized" => genes(model))
```

Comment:
the condition actually belongs here, here you want to check that gene_product_mass_groups is indeed nothing before the assignment. OTOH if you tripped over this I guess we should just completely erase the third "too default" argument and replace it with some function that makes the enzymes "all uncategorized"

Reply:
I actually want to remove all the default behavior. The user must specify at least something and it needs to be non-contradictory